src/pages/previous_questions.js

```diff
@@ -4,25 +4,31 @@ import { GET } from "../utils/fetch";
 import { useEffect, useState } from "react";
 
 async function getQuestions() {
-  const weekly_questions = await GET("weekly_questions");
-  const questions = [];
-  for (const value in weekly_questions) {
-    questions.push({
-      id: weekly_questions[value].question_id,
-      body: weekly_questions[value].question.body,
-      time: new Date(weekly_questions[value].week),
-    });
-  }
+  const weeklyQuestions = await GET("weekly_questions");
+  const questions = weeklyQuestions.map((value) => {
+    return {
```

Comment:
This is not a huge problem, but you are building both the BE and the FE, why don't they use the same names, so this is not needed?

Reply:
Sorry I didn't really understand, could you explain again?

Reply:
This `map` is only needed because the structure of the data is different between what the API provides and what the FE needs, right?

Reply:
The FE here wants all the questions that have been weekly_questions, then the answers to that questions and inside of that we want to know the user that made that answer. The API provides that, a list of weekly questions, so the FE just needs to go through each weekly question with the use of map and save the information on the questions variable that way it holds all the information in one single request. Now the API provides more information than what we needed, so the use of this map helps to structure the info as we want. This way, we make this route useful for other uses. Isn't this correct? Should I make the info the API returns really specific for a single request of the FE?

Reply:
Hmm ok, I see, yea since it's for this app it makes sense, thanks!